client/verta/verta/_registry/modelversions.py

```diff
@@ -29,5 +29,7 @@ def _call_back_end(self, msg):
     def _create_element(self, msg):
         return RegisteredModelVersion(self._conn, self._conf, msg)
 
-    def with_model(self, registered_model=None):
-        raise NotImplementedError
+    def with_model(self, registered_model):
+        new_list = copy.deepcopy(self)
+        new_list._registered_model = registered_model
```

Comment:
This should look more like [`Experiments.with_project()`](https://github.com/VertaAI/modeldb/blob/5cce4c4/client/verta/verta/_tracking/experiments.py#L45-L48)

Something like
```python
if registered_model:
    new_list._msg.id.registered_model_id = registered_model.id
...
```
But check [the protos](https://github.com/VertaAI/protos-all/blob/master/protos/protos/private/registry/RegistryService.proto#L171) (`new_list._msg` is `FindModelVersionRequest`) to verify the right field is assigned the ID.

Reply:
done